combat/combatSystem.cpp

```diff
@@ -373,6 +378,144 @@ int Monster::getWeaponSkill(const std::shared_ptr<Object>  weapon) const {
     return(weaponSkill);
 }
 
+//**********************************************************************
+//                      getClassWeaponskillBonus
+//**********************************************************************
+
+int Player::getClassWeaponskillBonus(const std::shared_ptr<Object>  weapon) const {
+    int cBonus = 0;
+
+    std::string weaponType;
+    if(weapon)
+        weaponType = weapon->getWeaponType();
+    else
+        weaponType = getUnarmedWeaponSkill();
+
+    // we're very confused about what type of weapon this is
+    if(weaponType.empty())
+        weaponType = "bare-hand";
+
+    if (getClass() == CreatureClass::CLERIC) {
+        switch (getDeity()) {
+        case MARA:
+            if (weaponType == "bow")
+                cBonus = 10;
+            break;
+        case LINOTHAN:
+            if (weaponType == "great-sword")
+                cBonus = 20;
+            break;
+        case ARACHNUS:
+            if (weaponType == "whip")
+                cBonus = 20;
+            break;
+        default:
+            break;
+        }
+    }
+
+    return(cBonus);
+}
+
+//**********************************************************************
+//                      getRacialWeaponskillBonus
+//**********************************************************************
+
+int Player::getRacialWeaponskillBonus(const std::shared_ptr<Object>  weapon) const {
```

Comment:
```suggestion
int Player::getRacialWeaponSkillBonus(const std::shared_ptr<Object>  weapon) const {
```
References would have to be updated as well

Reply:
new function so never been called til now